apps/storybook/src/Select.stories.tsx

```diff
@@ -344,12 +344,15 @@ export const TruncateMiddleText: Story<SelectProps<string>> = (args) => {
         <MenuItem>
           <MiddleTextTruncation
             text={option.label}
-            textRenderer={textRenderer}
+            textRenderer={textRenderer(option.label)}
```

Comment:
That is looking really weird :/
Could have something like `textRenderer={useCallback((truncatedText) => textRenderer(truncatedText, option.label))}` ?
Or ok..  you made your point and return the original text :D I think @rohan-kadkol is also voting to return both truncated and original text..

Reply:
Yes, thanks! Or maybe something like `textRenderer={textRenderer}`, where `textRenderer` is defined as

```ts
const textRenderer = useCallback(
  (originalText: string, truncatedText: string) => (
    <span title={truncatedText !== originalText ? originalText : undefined}>
      {truncatedText}
    </span>
  ),
  [],
);
```

and `MiddleTextTruncationProps`'s `textRenderer()` has `originalText` and `truncatedText`

```ts
export type MiddleTextTruncationProps = {
  ...

  textRenderer?: (originalText: string, truncatedText: string) => React.ReactNode;
};
```

and then when returning in the component, we pass `text` and `truncatedText` to `textRenderer()`

```tsx
return (
    <span
      ...
    >
      {textRenderer ? textRenderer(text, truncatedText) : truncatedText}
    </span>
  );
```

I feel with this way, users can easily get access to the `originalText` in `textRenderer()` and display what they want accordingly.

@veekeys What do you recommend?

Reply:
yeah.. that was the same as it was before done by @andriusgv 😄 
So as seems you both tend to agree to have original text passed, I guess we can do that and be done :)